### PR TITLE
Move to .NET SDK 5.0

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

**Comment by sgryphon** [comment]
Are we using any features from 5.0 ?   If not needed, do we want to stay compatible with the lowest version possible?

If we do go to 5.0, note that it is not LTS (like 3.1 LTS), so will go out of support relatively soon. So long as there is commitment to update to the next LTS (will be 6.0 LTS, even numbers, in Nov 2021).

**Comment by Mpdreamz** [comment]
The SDK is for local development only, in terms of TFM's we ship nothing should change. 

We typically jump on the latest greatest so once 6.0 comes out we'll jump to that SDK too. 

In terms of what we use nothing yet but I can imagine the models shifting to `record` at some point in the future for instance.

**Comment by sgryphon** [comment]
Seems okay apart from the unit test error in the BenchmarkIntegrationTests (that test is also failing locally for me)

**Comment by russcam** [comment]
I've fixed the benchmark test, which is related to https://github.com/dotnet/BenchmarkDotNet/issues/1452.

@Mpdreamz I left a couple of comments from review

File: .ci/docker/sdk-linux/Dockerfile [hunk 1]
```diff
@@ -1 +1 @@
-FROM mcr.microsoft.com/dotnet/core/sdk:3.1.202
+FROM mcr.microsoft.com/dotnet/core/sdk:5.0.100
```
[the remark this answers is not in the document]
Comment (russcam) [reply .ci/docker/sdk-linux/Dockerfile:1]: done!

**Comment by russcam** [comment]
jenkins run tests

**Pull request merged**